File: Python3/base_7.py
class Solution:
    def convertToBase7(self, num: int) -> str:
        if num == 0: return "0"
        
        result = ""
        n = abs(num)
        
        while n > 0:
            result = str(n % 7) + result
            n //= 7
        
        if num < 0 : result = "-" + result
            
        return result

File: C++/base_7.cpp
class Solution {
public:
    string convertToBase7(int num) {
        if(num == 0) return "0";
        
        string result = "";
        int n = abs(num);
        
        while(n > 0) {
            result = to_string(n % 7) + result;
            n /= 7;
        }
        
        if(num < 0) result = "-" + result;
            
        return result;
        
    }
};
